File: Problems/0203_Remove_Linked_List_Elements/Project_CS/Remove_Linked_List_Elements.cs
using System;

// Definition for singly-linked list.

public class ListNode {
    public int val;
    public ListNode next;
    public ListNode(int x) {
        val = x;
        next = null;
    }
}

public class Solution {
    public ListNode RemoveElements(ListNode head, int val)
    {
        ListNode dummy = new ListNode(-1);
        ListNode curr = dummy;

        while (head != null) {
            if (head.val != val) {
                curr.next = head;
                curr = curr.next;
                head = head.next;
            }
            else
                head = head.next;
        }

        curr.next = null;
        return dummy.next;
    }

    public int[] str_to_int_array(string s)
    {
        if (s.Length <= 0)
            return null;

        string[] flds = s.Split(',');
        int[] nums = new int[flds.Length];

        if (flds.Length <= 0)
            return nums;

        for (int i = 0; i < nums.Length; ++i)
        {
            nums[i] = int.Parse(flds[i]);
        }

        return nums;
    }

    public string output_int_array(int[] nums)
    {
        if (nums.Length <= 0)
            return "";

        string resultStr = "[" +  nums[0].ToString();
 
        for (int i = 1; i < nums.Length; ++i)
        {
            resultStr += "," + nums[i].ToString();
        }

        return resultStr + "]";
    }

    public void Main(string args)
    {
        string[] flds = args.Replace(" ", "").Replace("[[", "").Replace("]]", "").Split("],[", StringSplitOptions.None);
        int[] nums = str_to_int_array(flds[0]);
        int k = int.Parse(flds[1]);

        Operate_ListNode ope = new Operate_ListNode();
        ListNode node = ope.set_ListNode(nums);
        Console.WriteLine("node = " + ope.output_ListNode(node));

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        ListNode result_node = RemoveElements(node, k);
        sw.Stop();

        Console.WriteLine("Result = " + ope.output_ListNode(result_node));
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0200_0299/0203_Remove_Linked_List_Elements/Project_Python3/Remove_Linked_List_Elements.py
# coding: utf-8

import os
import sys
import time

from ListNode.ListNode import ListNode
from ListNode.OperateListNode import OperateListNode

class Solution:
    def removeElements(self, head, val):
        dummy = ListNode(-1)
        curr = dummy
        while head:
            if head.val != val:
                curr.next = head
                curr = curr.next
                head = head.next
            else:
                head = head.next
        curr.next = None
        return dummy.next

    def removeElements_work(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        temp_node = head
        while temp_node is not None:
            if temp_node.next is not None:
                if temp_node.next.val == val:
                    temp_node.next = temp_node.next.next
            temp_node = temp_node.next
        return head

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    str_args = temp.replace("\"","").replace(" ","").replace("[[","").replace("]]","").rstrip()
    flds = str_args.split("],[")

    ope_l = OperateListNode()
    head = ope_l.createListNode(flds[0])
    print("nodes = {0}".format(ope_l.ListNodeToString(head)))

    val = int(flds[1])
    print("val = {0:d}".format(val))

    sl = Solution()
    time0 = time.time()

    result = sl.removeElements(head, val)

    time1 = time.time()

    print("result = {0}".format(ope_l.ListNodeToString(result)))
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
